geniza/corpus/models.py

```diff
@@ -1,29 +1,41 @@
 from django.db import models
 
 
-class Library(models.Model):
-    '''Library or archive that holds Geniza fragments'''
-    name = models.CharField(max_length=255, unique=True)
+class Collection(models.Model):
+    '''Collection at a library that holds Geniza fragments'''
+    library = models.CharField(max_length=255)
     abbrev = models.CharField('Abbreviation', max_length=255, unique=True)
-    url = models.URLField('URL', blank=True)
+    collection = models.CharField(
+        max_length=255, blank=True,
+        help_text='Collection name, if different than Library')
+    location = models.CharField(
+        max_length=255, help_text='Current location of the collection')
 
     class Meta:
-        verbose_name_plural = 'Libraries'
-        ordering = ['name']
+        ordering = ['abbrev']
+        constraints = [
+            models.UniqueConstraint(fields=['library', 'collection'],
+                                    name='unique_library_collection')
+        ]
 
     def __str__(self):
-        return self.name
+        return self.abbrev
+
 
 class LanguageScript(models.Model):
     '''Combination language and script'''
     language = models.CharField(max_length=255)
     script = models.CharField(max_length=255)
-    display_name = models.CharField(max_length=255, blank=True, unique=True, null=True, 
+    display_name = models.CharField(
+        max_length=255, blank=True, unique=True, null=True,
         help_text="Option to override the autogenerated language-script name")
 
     class Meta:
         ordering = ['language']
-        constraints = [models.UniqueConstraint(fields=['language', 'script'], name='unique_language_script')]
+        constraints = [
+            models.UniqueConstraint(fields=['language', 'script'],
+                                    name='unique_language_script')
```

Comment:
whoa, I've never seen this `constraints` API before! since you defined uniqueness here, should be trivial to also implement `get_natural_key()` – just a thought

Reply:
This was new to me too! Credit to @kmcelwee for this one, I would have just used `unique_together`, but apparently that's deprecated now and might go away eventually.